fetch.bs

```diff
@@ -1121,6 +1111,46 @@ or "<code>worker</code>".
  <li><p>Return <var>newRequest</var>.
 </ol>
 
+<p>To <dfn export for=request id=concept-request-upload-body>upload body</dfn> of a
+<a for=/>request</a> <var>request</var>, run these steps:
+
+<ol>
+ <li>Let <var>body</var> be <var>request</var>'s <a for=request>body</a>.
+
+ <li><p>If <var>body</var> is null, then <a>queue a fetch task</a> on <var>request</var> to
+ <a>process request end-of-body</a> for <var>request</var> and abort these steps.
+
+ <li>
+  <p>Let <var>read</var> be the result of <a lt="read a chunk" for=ReadableStream>reading a
+  chunk</a> from <var>body</var>'s <a for=body>stream</a>.
+
+  <ul>
+   <li>
+    <p>When <var>read</var> is fulfilled with an object whose <code>done</code>
+    property is false and whose <code>value</code> property is a
+    <code>Uint8Array</code> object, run these substeps:
+
+    <ol>
+     <li><p>Let <var>bytes</var> be the bytes represented by the <code>Uint8Array</code> object.
+
+     <li><p>Upload <var>bytes</var>.
+
+     <li><p>Increase <var>body</var>'s <a for=body>transmitted bytes</a> with <var>bytes</var>'
+     length.
+
+     <li><p>Run the above step again.
+    </ol>
+
+   <li><p>When <var>read</var> is fulfilled with an object whose <code>done</code>
+   property is true, <a>queue a fetch task</a> on <var>request</var> to
+   <a>process request end-of-body</a> for <var>request</var>.
+
+   <li><p>When <var>read</var> is fulfilled with a value that matches with neither of the
+   above patterns, or <var>read</var> is rejected, <a lt=terminated for=fetch>terminate</a> the
+   ongoing fetch with reason <i>fatal</i>.
```

Comment:
This seems a little weird since we don't have a handle on the ongoing fetch and it's not passed to this algorithm. Perhaps it's okay for now though, but we should maybe file a follow up issue to see if we can fix that somehow.